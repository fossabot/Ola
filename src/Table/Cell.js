import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'

const TableCell = ({ variant, align, numeric, extraClass, children, ...props }) => {
  const Component = (variant === 'header') ? 'th' : 'td'
  return (
    <Component className={cx(numeric && 'ola-numeric', align && align !== 'left' && `ola-${align}`, extraClass)} {...props}>
      {children}
    </Component>
  )
}

TableCell.defaultProps = {
  align: 'left',
  numeric: false,
  variant: 'body'
}

TableCell.propTypes = {
  /** Extra className */
  extraClass: PT.string,
  /** Content align */
  align: PT.oneOf(['left', 'center', 'right']),
  /** Is numeric content */
  numeric: PT.bool,
  /** Variant */
  variant: PT.oneOf(['header', 'body']),
  /** Childen nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ])
}

export { TableCell }
