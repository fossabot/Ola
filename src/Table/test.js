import React from 'react'
import { Table } from './'
import { TableCell } from './Cell'
import renderer from 'react-test-renderer'


it('Default Table', () => {
  const tree = renderer
    .create(
      <Table>
        <tbody></tbody>
      </Table>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Table with extra class', () => {
  const tree = renderer
    .create(
      <Table extraClass="extra-class">
        <tbody></tbody>
      </Table>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Add caption', () => {
  const tree = renderer
    .create(
      <Table caption="test caption">
        <tbody></tbody>
      </Table>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Can be stiky', () => {
  const tree = renderer
    .create(
      <Table sticky>
        <tbody></tbody>
      </Table>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Default Cell', () => {
  const tree = renderer
    .create(
      <TableCell>Test content</TableCell>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Header Cell', () => {
  const tree = renderer
    .create(
      <TableCell variant="header">Content</TableCell>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Numeric Cell', () => {
  const tree = renderer
    .create(
      <TableCell numeric>8</TableCell>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('Align center Cell', () => {
  const tree = renderer
    .create(
      <TableCell align="center">Test content</TableCell>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
