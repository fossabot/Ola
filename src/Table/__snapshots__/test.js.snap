// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add caption 1`] = `
<div
  className="ola_table-container"
>
  <table
    className="ola_table"
  >
    <caption>
      test caption
    </caption>
    <tbody />
  </table>
</div>
`;

exports[`Align center Cell 1`] = `
<td
  className="ola_tableCell is-center"
  style={
    Object {
      "--borderColor": undefined,
      "--table-background": undefined,
    }
  }
>
  Test content
</td>
`;

exports[`Can be responsive 1`] = `
<div
  className="ola_table-container is-responsive"
>
  <table
    className="ola_table"
  >
    <tbody />
  </table>
</div>
`;

exports[`Can be stiky 1`] = `
<div
  className="ola_table-container"
>
  <table
    className="ola_table is-sticky"
  >
    <tbody />
  </table>
</div>
`;

exports[`ColSpan Cell 1`] = `
<td
  className="ola_tableCell is-left"
  colSpan={8}
  style={
    Object {
      "--borderColor": undefined,
      "--table-background": undefined,
    }
  }
/>
`;

exports[`Default Cell 1`] = `
<td
  className="ola_tableCell is-left"
  style={
    Object {
      "--borderColor": undefined,
      "--table-background": undefined,
    }
  }
>
  Test content
</td>
`;

exports[`Default Table 1`] = `
<div
  className="ola_table-container"
>
  <table
    className="ola_table"
  >
    <tbody />
  </table>
</div>
`;

exports[`Empty Cell 1`] = `
<td
  className="ola_tableCell is-left"
  style={
    Object {
      "--borderColor": undefined,
      "--table-background": undefined,
    }
  }
/>
`;

exports[`Header Cell 1`] = `
<th
  className="ola_tableCell is-left"
  style={
    Object {
      "--borderColor": undefined,
      "--table-background": undefined,
    }
  }
>
  Content
</th>
`;

exports[`Numeric Cell 1`] = `
<td
  className="ola_tableCell is-numeric"
  style={
    Object {
      "--borderColor": undefined,
      "--table-background": undefined,
    }
  }
>
  8
</td>
`;

exports[`Table row 1`] = `
<tr
  className="ola_tableRow is-selectable"
>
  <td
    className="ola_tableCell is-check"
    style={
      Object {
        "--borderColor": undefined,
        "--table-background": undefined,
      }
    }
  >
    <label
      className="ola_check"
    >
      <input
        className="ola_check-input"
        type="checkbox"
      />
      <div
        className="ola_check-label"
      />
    </label>
  </td>
  <td
    className="ola_tableCell is-left"
    style={
      Object {
        "--borderColor": undefined,
        "--table-background": undefined,
      }
    }
  >
    First column
  </td>
</tr>
`;

exports[`Table with extra class 1`] = `
<div
  className="ola_table-container"
>
  <table
    className="ola_table extra-class"
  >
    <tbody />
  </table>
</div>
`;
